src/controller/proxy/controller.go

```diff
@@ -94,22 +92,44 @@ func (c *controller) UseLocalBlob(ctx context.Context, art lib.ArtifactInfo) boo
 	return exist
 }
 
-func (c *controller) UseLocalManifest(ctx context.Context, art lib.ArtifactInfo) bool {
-	if len(art.Digest) == 0 {
-		return false
+func (c *controller) UseLocalManifest(ctx context.Context, art lib.ArtifactInfo, remote RemoteInterface) (bool, error) {
+	inLocal, lDig, err := c.local.ManifestExist(ctx, art)
```

Comment:
In `ProxyManifest` the manifest was read again, so would it be better if you check whether the digests are the same in `waitAndPushManifest` ?

Reply:
Update